include/picongpu/plugins/adios/ADIOSWriter.hpp

```diff
@@ -331,6 +331,9 @@ namespace picongpu
 #ifndef __CUDA_ARCH__
                     DataConnector& dc = Environment<simDim>::get().DataConnector();
 
+                    // Skip PML fields when PML is not enabled
+                    if(!dc.hasId(T_Field::getName()))
```

Comment:
This looks dangerous, it looks like all fields which are not registered in the data connector will be skipped.
Do we have maybe a better way to explicitly only skip PML fields and not all unknown?

Maybe:

```
if(T_Field::getName() == "pmlFields" && !dc.hasId(T_Field::getName())
```

Reply:
Yes, so here we have an issue that some part, the typelist of fields in this case, is compile-time, and another part is run-time. I would not like to have any PML information here. Could make a trait, is it good with you?

Reply:
Yes whatever fits best, as long as it avoids skipping IO if an unknown field is added somewhere in param or unitless.
This would avoid that we search in the future why our configured data is not written.

Reply:
@psychocoderHPC as an alternative. I could make it so that the convolutional fields are always created and regiatered in the data connector. PML implementation will create them of the actual size, exponential implementation with size 0 (if that is allowed everywhere) or 1. Then in checkpointing and output there is no need to know about them or treat them separately. So that is better in terms of interdependencies in the code. On the other hand, it could make the output files more confusing. What do you think?

Reply:
I think if we can avoid creating PML helper field and skip dumping this would be better for IO scaling. Even writing out a dataset with zero entries can result in MPI calls (depending of the IO backend) and meta data is written.

Reply:
> Now we also have another place for separate treatment: field reset after the window moves. That part also now checks if the PML fields are registered in the data connector. Kinda like output implementation as in PR currently does.

I would keep checking. Maybe (not in this PR) we should think about refactoring our DataConnector a little bit and additional information if data should be part of a checkpoint, by adding a flag. We already talked about making checkpointing more explicit maybe this is falling into the same topic.

Reply:
Maybe you can make a doxygen comment out of the comment above and add `//!@todo`

Reply:
Could you please add the filter trait or via name to finalize this PR.

Reply:
Done

Reply:
@sbastrakov Do you test your PR with a simulation where some GPUs do not have the PML field? The problem is that IO is a collective operation and all ranks must participate.
IMO if some ranks do not have this additional field than not all ranks will do the IO. 
It depends on the implementation. Do you register `pmlFields` with a size zero for ranks which will not be calculating a simulation boundery?

Reply:
When PML is enabled, all GPUs have it. And all of them have PML fields of the max size that a domain can have. It was like that even before this PR. The reason is that "central" GPUs can become "border" ones due to moving window, and we need to preallocate memory at simulation start. So I do not expect this PR to change anything there. But will recheck

Reply:
Thx for the clarification.